### PR TITLE
Add memory info to docs.

### PR DESCRIPTION
**Pull request by htroisi**
Signed-off-by: Hannah Troisi <htroisi@pixielabs.ai>

### CODE REVIEW

File: content/en/02-installing-pixie/01-requirements.md [hunk 1]
```diff
@@ -69,23 +62,42 @@ Pixie runs on Linux nodes only.
...
+| :------ | :---------------- |
+| x86-64  | Supported         |
+| ARM     | Not Supported     |
 
 ## Network Traffic
 
 Pixie's [Vizier](/reference/architecture/#vizier) module sends outgoing HTTPS/2 requests to Pixie's [Cloud](/reference/architecture/#cloud) on port `443`.
 
 Your cluster's telemetry data flows through Pixie's Cloud via a reverse proxy as encrypted traffic without any persistence. This allows users to access data without being in the same VPC/network as the cluster. Pixie offers [end-to-end encryption](/about-pixie/faq#data-collection-how-does-pixie-secure-its-data) for telemetry data in flight.
+re
```
Comment (nserrino) [inline content/en/02-installing-pixie/01-requirements.md]: remove

File: content/en/02-installing-pixie/01-requirements.md [hunk 2]
```diff
@@ -69,23 +62,42 @@ Pixie runs on Linux nodes only.
...
+
+## Memory
+
+Memory requirements for your cluster nodes are as follows:
+
+| Minimum   | Notes                                                                  |
+| :-------- | :--------------------------------------------------------------------- |
+| 1GiB      | To accommodate application pods, 4GiB+ total per node is recommended.  |
```
Comment (nserrino) [inline content/en/02-installing-pixie/01-requirements.md]: People get confused about total memory vs Pixie memory. I think it's fairly clear but let's be even more clear. 

To accommodate application pods, we recommend using no more than 25% of the nodes' total memory for Pixie.

File: content/en/01-about-pixie/06-troubleshooting.md [hunk 3]
```diff
@@ -113,7 +113,9 @@ Pixie's [continuous profiler](/tutorials/pixie-101/profiler/) currently supports
 
 This is expected behavior. Pixie stores the data it collects in-memory on the nodes in your cluster; data is not sent to any centralized backend cloud outside of the cluster. So what you are observing is simply the data that it is collecting.
```
Comment (nserrino) [inline content/en/01-about-pixie/06-troubleshooting.md:115]: Let's add an FAQ about OOMKilling.

<paraphrased>
Question: why is pixie oomkilling?
Answer:  this happens when Pixie is using more memory than the limit. Try increasing the limit, or reducing the amount of data stored in the table store. The amount of memory needed for Pixie depends on the application, but a minimum of 1Gi is required.

**Review by nserrino — COMMENTED** [review]
I think we might want to add a memory specific admin page, Tuning Vizier memory usage or something like that. and link to it at the end of the install guides

File: content/en/01-about-pixie/06-troubleshooting.md (hunk 3 above)
Comment (htroisi) [reply content/en/01-about-pixie/06-troubleshooting.md:115]: I'm going to add this to the Troubleshooting page (not FAQ).

**Comment by htroisi** [comment]
@nserrino - I'm not sure what we would put in the "Tuning Vizier Memory Usage" page that isn't already in the Deploy Options page. Would this make more sense to add once we have the memory sizing PxL script?

File: content/en/01-about-pixie/06-troubleshooting.md [hunk 4]
```diff
@@ -22,6 +22,7 @@ This page describes how to troubleshoot Pixie. We frequently answer questions on
 - [Why can’t I see data?](#troubleshooting-operation-why-can't-i-see-data)
 - [Why can’t I see application profiles / flamegraphs for my pod / node?](#troubleshooting-operation-why-can't-i-see-application-profiles-flamegraphs-for-my-pod-node)
 - [Why is the vizier-pem pod’s memory increasing?](#troubleshooting-operation-why-is-the-vizier-pem-pod's-memory-increasing)
+- [Why is the Pixie oomkilling?](#troubleshooting-operation-why-is-pixie-oomkilling)
```
Comment (nserrino) [inline content/en/01-about-pixie/06-troubleshooting.md]: being OOMKilled

File: content/en/01-about-pixie/06-troubleshooting.md [hunk 5]
```diff
@@ -113,7 +114,13 @@ Pixie's [continuous profiler](/tutorials/pixie-101/profiler/) currently supports
 
 This is expected behavior. Pixie stores the data it collects in-memory on the nodes in your cluster; data is not sent to any centralized backend cloud outside of the cluster. So what you are observing is simply the data that it is collecting.
 
-Pixie has a minimum 1GiB memory requirement per node. The default deployment is 2GiB of memory. This limit can be configured with the `--pem_memory_limit` flag when deploying Pixie. Using a value less than 1GiB is not currently recommended.
+When Pixie is first deployed, you can expect to see an increase in memory utilization over time. This is because the temporary data store will go from storing 0B of data to reaching its cap of 1.25G of data (in the default deployment). Once this cap is hit, old data is expired and the memory utilization shouldn’t increase anymore.
+
+Pixie has a minimum 1GiB memory requirement per node. The default deployment is 2GiB of memory. To learn how to configure Pixie's memory usage, see the [Tuning Memory Usage](/reference/admin/tuning-mem-usage/) page.
+
+### Why is Pixie oomkilling?
```
Comment (nserrino) [inline content/en/01-about-pixie/06-troubleshooting.md]: being OOM killed

File: content/en/01-about-pixie/06-troubleshooting.md [hunk 6]
```diff
@@ -113,7 +114,13 @@ Pixie's [continuous profiler](/tutorials/pixie-101/profiler/) currently supports
...
+
+This happens when Pixie is using more memory than the limit. Try [increasing the limit](/reference/admin/deploy-options/#configuring-pixie's-memory-usage-setting-the-memory-limit), or [reducing the amount of data stored](/reference/admin/deploy-options/#configuring-pixie's-memory-usage-setting-the-data-table-storage-memory-limit) in the table store. The amount of memory needed for Pixie depends on the application, but a minimum of [1Gi](/installing-pixie/requirements/#memory) is required.
```
Comment (nserrino) [inline content/en/01-about-pixie/06-troubleshooting.md]: Add a sentence that we generally recommend 2Gi

File: content/en/02-installing-pixie/01-requirements.md [hunk 7]
```diff
@@ -5,14 +5,24 @@ metaDescription: "Resource requirements to install Pixie"
...
 
 Please refer to the [install guides](/installing-pixie/install-guides/) for information on how to install Pixie to your K8s cluster.
 
 ## Kubernetes
 
 Kubernetes `v1.16+` is required.
 
+The following tables list Kubernetes environments that have been tested with Pixie.
+
+<Alert variant="outlined" severity="info">Pixie may work on other K8s environments. If you find an unlisted K8s environment that works well, please let us know by opening a GitHub issue or by creating a pull request that updates <a href="https://github.com/pixie-io/docs.px.dev/blob/main/content/en/02-installing-pixie/01-requirements.md">this doc</a>.</Alert>
```
Comment (nserrino) [inline content/en/02-installing-pixie/01-requirements.md]: I would try to make changes like this in separate PRs since iit gets confusing with unrelated stuff in the same diff

File: content/en/02-installing-pixie/01-requirements.md [hunk 8]
```diff
@@ -69,23 +62,41 @@ Pixie runs on Linux nodes only.
 | Linux   | Supported       | v4.14+            |
 | Windows | Not Supported   | Not in roadmap    |
 
-### Linux Distributions
+### Linux Distribution
```
Comment (nserrino) [inline content/en/02-installing-pixie/01-requirements.md]: same feedback here - let's try to keep unrelated changes in a separate diff, it's harder to review bigger diffs that have a lot going on

File: content/en/05-reference/01-admin/04-deploy-options.md [hunk 9]
```diff
@@ -7,52 +7,150 @@ redirect_from:
...
-helm install pixie pixie-operator/pixie-operator-chart --set deployKey=<deploy-key-goes-here> --set clusterName=<cluster-name> --namespace pl --create-namespace
---dataAccess=(Full|Restricted)
+
+### Setting the memory request
+
+When deploying Pixie, you can specify a different memory [request](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits) than limit. This flag is useful for clusters in which the PEM pods are pending because there is not enough memory for them to be scheduled on the node.
+
+<Alert variant="outlined" severity="info">Please note that PEM memory request must be less than or equal to the PEM memory limit.</Alert>
+
+To set Pixie's memory request when deploying with the [Pixie CLI](/installing-pixie/install-schemes/cli/), use the `--pem_memory_request` flag:
```
Comment (nserrino) [inline content/en/05-reference/01-admin/04-deploy-options.md:106]: Can you note that by default, the limit and the request will be the same unless they are both specified?

File: content/en/05-reference/01-admin/09-tuning-mem-usage.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
+---
+title: "Tuning Memory Usage"
+metaTitle: "Reference | Admin | Tuning Memory Usage"
+metaDescription: "Tuning Pixie vizier-pem memory usage"
+order: 9
+---
+
+One of Pixie's unique features is that it collects, stores and queries all telemetry data *locally in your cluster*. Unlike other observability platforms, no telemetry data is transferred to a centralized backend outside of the cluster.
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-tuning-mem-usage.md]: I would drop Unlike other observability platforms but keep the rest of the sentence

Comment (nserrino) [reply content/en/05-reference/01-admin/09-tuning-mem-usage.md]: We may also want to add, "unless you choose to configure a plugin or export using the API" as a way to hype those features, and link out to them

File: content/en/05-reference/01-admin/09-tuning-mem-usage.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+The component responsible for collecting and storing the telemetry data in your cluster is called a [PEM](/reference/architecture/#vizier-components) (Pixie Edge Module).
+When Pixie is installed, a `vizier-pem` data collector pod is deployed to each node in your cluster via a DaemonSet. By default, `vizier-pem` pods have a 2Gi memory limit (but this can be configured to be lower).
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-tuning-mem-usage.md]: `2Gi` to make it pop

Comment (nserrino) [reply content/en/05-reference/01-admin/09-tuning-mem-usage.md]: this can be configured to be lower or higher

File: content/en/05-reference/01-admin/09-tuning-mem-usage.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+By default, `vizier-pem` pods reserve 60% of their allocated memory for short-term data storage (leaving the other 40% for the collection). For the default 2Gi memory limit, `vizier-pem` pods will reserve 1.2Gi memory for data storage.
+
+For developers with [long-term data storage](/about-pixie/faq/#how-do-i...-how-do-i-export-data-from-the-pixie-platform-import-data), Pixie can be configured to allocate less memory for data storage.
+
+### If Pixie stores data locally, why does it have a cloud component?
+
+Pixie's [cloud](/reference/architecture/#cloud) component serves Pixie's API and UI to make it easy for developers to access telemetry data without being in the same network as the cluster. When you query the Pixie platform using the CLI, Live UI or API, your telemetry data flows through Pixie's Cloud via a reverse proxy as E2E encrypted traffic without any persistence.
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-tuning-mem-usage.md]: Link out to docs/diagrams/articles about the encryption stuff. E2E -> end-to-end. Also worth mentioning that Pixie cloud manages users, etc

Comment (nserrino) [reply content/en/05-reference/01-admin/09-tuning-mem-usage.md]: Also, not for this diff, but we should add more diagrams to https://deploy-preview-178--docs-px-dev.netlify.app/reference/architecture/#cloud

File: content/en/05-reference/01-admin/09-tuning-mem-usage.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+## Configuring Pixie's memory usage
+
+The proper setting for Pixie's memory usage is dependent on your cluster / environment.
+
+For most clusters, we recommend that you initially try deploying Pixie with the default (2Gi) memory limit. To accommodate your application pods, we recommend that this amount be no more than 25% of your nodes' total memory. In other words, with the default configuration your nodes should have 4Gi+ total memory.
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-tuning-mem-usage.md]: `2Gi` (here and other places)

Comment (nserrino) [reply content/en/05-reference/01-admin/09-tuning-mem-usage.md]: 2Gi is 50% of 4Gi. So you might just want to say, use a 1Gi memory limit in Pixie if you have 4Gi total on the node.

Comment (nserrino) [reply content/en/05-reference/01-admin/09-tuning-mem-usage.md]: Let's make a bulleted list of the top level takeaways for each of these sections for the skimmers. There can also be prose but I think bulleted with the tl;dr will be useful.

File: content/en/05-reference/01-admin/09-tuning-mem-usage.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+To learn how to set Pixie's memory request when deploying, see the [Deploy Options](/reference/admin/deploy-options#configuring-pixie's-memory-usage-setting-the-memory-request) page.
+
+### How can I deploy Pixie with a particular memory limit?
+
+If your `vizier-pem` pods still fail to schedule during installation, you can try deploying with a lower memory limit. The default memory limit is 2Gi per PEM. The lowest recommended value is [1Gi](/installing-pixie/requirements/#memory) per PEM. 1Gi is not a suitable limit for a cluster with high throughput, but may be suitable for a small cluster with limited resources.
+
+To learn how to set Pixie's memory limit when deploying, see the [Deploy Options](/reference/admin/deploy-options#configuring-pixie's-memory-usage-setting-the-memory-limit) page.
+
+### (Advanced) How can I configure `vizie-pem` pods to reserve more or less memory for data storage?
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-tuning-mem-usage.md]: Bump advanced to the end

File: content/en/05-reference/01-admin/09-tuning-mem-usage.md [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+As discussed [above](#configuring-pixie's-memory-usage-how-does-pixie-use-memory), Pixie's `vizier-pem` data collector pods use memory for two main purposes: collecting data and storing data. By default, `vizier-pem` pods reserve 60% of their allocated memory for short-term data storage (leaving the other 40% for the collection). For the default 2Gi memory limit, this means that a `vizier-pem` pod will reserve 1.2Gi memory for data storage.
+
+To reduce Pixie's memory footprint, Developers with a [long-term data storage](/about-pixie/faq/#how-do-i...-how-do-i-export-data-from-the-pixie-platform-import-data) solution can configure Pixie to use less memory for short-term data storage on the node.
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-tuning-mem-usage.md]: Lower case Developers->developers

File: content/en/05-reference/01-admin/09-tuning-mem-usage.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+
+### Why did the `vizier-pem` pods fail to schedule?
+
+If your `vizier-pem` pods fail to schedule during installation, this is usually because there are node(s) in your cluster that do not have the requested amount of memory. In this situation, you can try reducing the memory [request](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits) down as far as 1Gi. However, keep in mind that if it is a particularly large or active cluster, 1Gi may not be sufficient for that particular cluster, and some PEM OOMKills might occur, so it might be good to set a higher limit than request, such as 2Gi for the limit. 1Gi limit deployments are recommended for smaller clusters with less traffic.
+
+To learn how to set Pixie's memory request when deploying, see the [Deploy Options](/reference/admin/deploy-options#configuring-pixie's-memory-usage-setting-the-memory-request) page.
+
+### How can I deploy Pixie with a particular memory limit?
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-tuning-mem-usage.md:44]: Let's have this one come before the fail to schedule

**Review by nserrino — CHANGES_REQUESTED** [review]

File: content/en/01-about-pixie/06-troubleshooting.md [hunk 17]
```diff
@@ -113,7 +114,13 @@ Pixie's [continuous profiler](/tutorials/pixie-101/profiler/) currently supports
...
+When Pixie is first deployed, you can expect to see an increase in memory utilization over time. This is because the temporary data store will go from storing 0B of data to reaching its cap of 1.25G of data (in the default deployment). Once this cap is hit, old data is expired and the memory utilization shouldn’t increase anymore.
+
+Pixie has a minimum 1GiB memory requirement per node. The default deployment is 2GiB of memory. To learn how to configure Pixie's memory usage, see the [Tuning Memory Usage](/reference/admin/tuning-mem-usage/) page.
+
+### Why is Pixie being OOMKilled?
+
+This happens when Pixie is using more memory than the limit. Try [increasing the limit](/reference/admin/deploy-options/#configuring-pixie's-memory-usage-setting-the-memory-limit), or [reducing the amount of data stored](/reference/admin/deploy-options/#configuring-pixie's-memory-usage-setting-the-data-table-storage-memory-limit) in the table store. The amount of memory needed for Pixie depends on the application, but a minimum of [1Gi](/installing-pixie/requirements/#memory) is required. However, for most clusters we recommend 2Gi.
```
Comment (nserrino) [inline content/en/01-about-pixie/06-troubleshooting.md]: might want to link Tuning memory usage here

**Review by nserrino — DISMISSED** [review]

**Review by nserrino — APPROVED** [review]
